File: JavaScript/leetcode_242.js
/* 
Given two strings s and t, return true if t is an anagram of s, and false otherwise.
*/

var isAnagram = function(s, t) {
    if (s.length !== t.length) return false;

    const count = Array(26).fill(0); // 26 letters in the alphabet

    for (let i = 0; i < s.length; i++) {
        count[s.charCodeAt(i) - 'a'.charCodeAt(0)] += 1;
        count[t.charCodeAt(i) - 'a'.charCodeAt(0)] -= 1;
    }

    return count.every(c => c === 0);
};

File: Python/leetcode_242.py
'''
Given two strings s and t, return true if t is an anagram of s
, and false otherwise.
'''


def is_anagram(s: str, t: str) -> bool:
    if len(s) != len(t):
        return False

    count = [0] * 26  # Assuming only lowercase letters

    for ch1, ch2 in zip(s, t):
        count[ord(ch1) - ord('a')] += 1
        count[ord(ch2) - ord('a')] -= 1

    return all(c == 0 for c in count)
